internal/containers/provider/docker.go

```diff
@@ -80,9 +109,33 @@ func (p *DockerProvider) CreateNetwork(ctx context.Context, name string) (string
 	return mode.ID, nil
 }
 
+func (p *DockerProvider) GetNetwork(ctx context.Context, name string) (string, error) {
+	existing, err := p.cli.NetworkList(ctx, types.NetworkListOptions{
+		Filters: filters.NewArgs(filters.Arg("name", fmt.Sprintf("^/?%s$", name))),
+	})
+	if err != nil {
+		return "", fmt.Errorf("listing networks: %w", err)
+	}
+
+	log.Info(ctx, fmt.Sprintf("networks existing: %v", existing))
+
+	if len(existing) == 0 {
+		return "", ErrNetworkNotFound
+	}
+
+	targetNetwork, err := p.cli.NetworkInspect(ctx, existing[0].ID, types.NetworkInspectOptions{})
+	if err != nil {
+		return "", fmt.Errorf("inspecting network %s: %w", name, err)
+	}
+
+	return targetNetwork.ID, nil
+}
+
+var ErrNetworkNotFound = fmt.Errorf("network not found")
```

Comment:
Any reason not to use `errors.New` here?

```suggestion
var ErrNetworkNotFound = errors.New("network not found")
```

Reply:
sorry, this is leftover. fixed!